src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java

```diff
@@ -83,11 +84,14 @@ public ReadDocument getNewDocument() {
     return doc;
   }
 
-  private QueryOuterClass.Query buildSelectQuery() {
+  private QueryOuterClass.Query buildSelectQuery(DBFilterBase.IDFilter idOverride) {
```

Comment:
Added _id filter with other conditions instead of filtering the data in memory. When the query to cassandra has partition key filter, it will only route the query to that specific range that has the partition. Since it's not a not run range query this should be faster to filter.

Reply:
We are reading document based on the filter in came in request and additional id filter. I don't see adding more conditions to this with the use cases we have. Not required to be collection unless we have use case for it. 
Renamed it the variable as `additionalIdFilter`.

Reply:
So it's not an override in fact? What happens if the original query already had id condition?

Reply:
Ah, this will be a problem, fixed it.